src/scripts/main.js

```diff
@@ -1 +1,78 @@
 'use strict';
+
+function notification(value) {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.classList = value[0];
+  div.innerText = value[1];
+  document.body.appendChild(div);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    const value = ['success', 'First promise was resolved'];
+
+    resolve(value);
+
+    e.preventDefault();
+  });
+
+  setTimeout(() => {
+    const value = ['error', 'First promise was rejected'];
+
+    reject(value);
+  }, 3000);
+});
+
+firstPromise
+  .then((value) => {
+    notification(value);
+  })
+
+  .catch((value) => {
+    notification(value);
+  });
+
+const secondPromise = new Promise((resolve) => {
+  const value = ['success', 'Second promise was resolved'];
+
+  document.addEventListener('click', (e) => {
+    resolve(value);
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    resolve(value);
+  });
+});
+
+secondPromise.then((value) => {
+  notification(value);
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let rightClick = false;
+  let leftClick = false;
+
+  document.addEventListener('click', (e) => {
+    leftClick = true;
+    both();
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    rightClick = true;
+    both();
+  });
```

Comment:
The third promise resolves when both a left-click and a right-click have occurred. However, the current implementation does not reset the `rightClick` and `leftClick` flags, which means the promise will resolve on any subsequent click after both have been set to true. Consider resetting these flags after resolving the promise to prevent unintended behavior.